src/mono/wasm/runtime/jiterpreter-opcodes.ts

```diff
@@ -0,0 +1,1816 @@
+    MINT_ICALL_P_V,
+    MINT_ICALL_P_P,
+    MINT_ICALL_PP_V,
+    MINT_ICALL_PP_P,
+    MINT_ICALL_PPP_V,
+    MINT_ICALL_PPP_P,
+    MINT_ICALL_PPPP_V,
+    MINT_ICALL_PPPP_P,
+    MINT_ICALL_PPPPP_V,
+    MINT_ICALL_PPPPP_P,
+    MINT_ICALL_PPPPPP_V,
+    MINT_ICALL_PPPPPP_P,
+    // FIXME: MintOp
+    MINT_JIT_CALL,
+    MINT_JIT_CALL2,
+
+    MINT_MONO_LDPTR,
+    MINT_MONO_SGEN_THREAD_INFO,
+    MINT_MONO_NEWOBJ,
+    MINT_MONO_RETOBJ,
+    MINT_MONO_ATOMIC_STORE_I4,
+    MINT_MONO_MEMORY_BARRIER,
+    MINT_MONO_EXCHANGE_I8,
+    MINT_MONO_LDDOMAIN,
+    MINT_MONO_ENABLE_GCTRANS,
+
+    MINT_SDB_INTR_LOC,
+    MINT_SDB_SEQ_POINT,
+    MINT_SDB_BREAKPOINT,
+    MINT_LD_DELEGATE_METHOD_PTR,
+
+    // Math intrinsics
+    // double
+    MINT_ASIN,
+    MINT_ASINH,
+    MINT_ACOS,
+    MINT_ACOSH,
+    MINT_ATAN,
+    MINT_ATANH,
+    MINT_ATAN2,
+    MINT_CEILING,
+    MINT_COS,
+    MINT_CBRT,
+    MINT_COSH,
+    MINT_EXP,
+    MINT_FMA,
+    MINT_FLOOR,
+    MINT_LOG,
+    MINT_LOG2,
+    MINT_LOG10,
+    MINT_POW,
+    MINT_SCALEB,
+    MINT_SIN,
+    MINT_SQRT,
+    MINT_SINH,
+    MINT_TAN,
+    MINT_TANH,
+    MINT_ABS,
+    MINT_MIN,
+    MINT_MAX,
+
+    // float. These must be kept in the same order as their double counterpart
+    MINT_ASINF,
+    MINT_ASINHF,
+    MINT_ACOSF,
+    MINT_ACOSHF,
+    MINT_ATANF,
+    MINT_ATANHF,
+    MINT_ATAN2F,
+    MINT_CEILINGF,
+    MINT_COSF,
+    MINT_CBRTF,
+    MINT_COSHF,
+    MINT_EXPF,
+    MINT_FMAF,
+    MINT_FLOORF,
+    MINT_LOGF,
+    MINT_LOG2F,
+    MINT_LOG10F,
+    MINT_POWF,
+    MINT_SCALEBF,
+    MINT_SINF,
+    MINT_SQRTF,
+    MINT_SINHF,
+    MINT_TANF,
+    MINT_TANHF,
+    MINT_ABSF,
+    MINT_MINF,
+    MINT_MAXF,
+
+    MINT_PROF_ENTER,
+    MINT_PROF_EXIT,
+    MINT_PROF_EXIT_VOID,
+    MINT_PROF_COVERAGE_STORE,
+
+    MINT_TIER_ENTER_METHOD,
+    MINT_TIER_PATCHPOINT,
+
+    MINT_INTRINS_ENUM_HASFLAG,
+    MINT_INTRINS_GET_HASHCODE,
+    MINT_INTRINS_GET_TYPE,
+    MINT_INTRINS_SPAN_CTOR,
+    MINT_INTRINS_UNSAFE_BYTE_OFFSET,
+    MINT_INTRINS_RUNTIMEHELPERS_OBJECT_HAS_COMPONENT_SIZE,
+    MINT_INTRINS_CLEAR_WITH_REFERENCES,
+    MINT_INTRINS_MARVIN_BLOCK,
+    MINT_INTRINS_ASCII_CHARS_TO_UPPERCASE,
+    MINT_INTRINS_MEMORYMARSHAL_GETARRAYDATAREF,
+    MINT_INTRINS_ORDINAL_IGNORE_CASE_ASCII,
+    MINT_INTRINS_64ORDINAL_IGNORE_CASE_ASCII,
+    MINT_INTRINS_U32_TO_DECSTR,
+    MINT_INTRINS_WIDEN_ASCII_TO_UTF16,
+
+    // TODO: Make this wasm only
+    MINT_TIER_PREPARE_JITERPRETER,
+    MINT_TIER_NOP_JITERPRETER,
+    MINT_TIER_ENTER_JITERPRETER,
+
+    MINT_LASTOP
+}
+
+export const enum MintOpArgType {
+	MintOpNoArgs = 0,
+	MintOpShortInt,
+	MintOpUShortInt,
+	MintOpInt,
+	MintOpLongInt,
+	MintOpFloat,
+	MintOpDouble,
+	MintOpBranch,
+	MintOpShortBranch,
+	MintOpSwitch,
+	MintOpMethodToken,
+	MintOpFieldToken,
+	MintOpClassToken,
+	MintOpTwoShorts,
+	MintOpTwoInts,
+	MintOpShortAndInt,
+	MintOpShortAndShortBranch,
+	MintOpPair2,
+	MintOpPair3,
+	MintOpPair4
```

Comment:
I guess numeric value has to match C code, right ? 
Would comment on both locations to keep it in sync help ?

Reply:
I think this enum won't change often, but I should definitely make sure there are more comments about keeping it in sync